client/src/pages/spectrogram/components/PluginsPane.tsx

```diff
@@ -95,70 +101,98 @@ export const PluginsPane = ({ cursorsEnabled, handleProcessTime, meta, setMeta }
           return;
         }
         if (data.data_output && data.data_output.length > 0) {
-          // just show the first output for now, 99% of plugins will have 0 or 1 IQ output anyway
-          const samples_base64 = data.data_output[0]['samples'];
-          const samples = convertBase64ToFloat32Array(samples_base64);
-          //const sample_rate = data.data_output[0]['sample_rate']; // Hz
-          //const center_freq = data.data_output[0]['center_freq']; // Hz
-          //const data_type = data.data_output[0]['data_type']; // assumes iq/cf32_le
-          setModalSamples(samples);
-
-          // create spectrogram out of all samples
-          const fftSize = 1024;
-          const numFfts = Math.floor(samples.length / 2 / fftSize);
-          const magnitudeMin = -40;
-          const magnitudeMax = -10;
-          const samples_typed = Float32Array.from(samples);
-
-          let startOfs = 0;
-          let newFftData = new Uint8ClampedArray(numFfts * fftSize * 4); // 4 because RGBA
-
-          // loop through each row
-          for (let i = 0; i < numFfts; i++) {
-            let samples_slice = samples_typed.slice(i * fftSize * 2, (i + 1) * fftSize * 2); // mult by 2 because this is int/floats not IQ samples
-
-            const f = new FFT(fftSize);
-            let out = f.createComplexArray(); // creates an empty array the length of fft.size*2
-            f.transform(out, samples_slice); // assumes input (2nd arg) is in form IQIQIQIQ and twice the length of fft.size
-
-            out = out.map((x) => x / fftSize); // divide by fftsize
-
-            // convert to magnitude
-            let magnitudes = new Array(out.length / 2);
-            for (let j = 0; j < out.length / 2; j++) {
-              magnitudes[j] = Math.sqrt(Math.pow(out[j * 2], 2) + Math.pow(out[j * 2 + 1], 2)); // take magnitude
-            }
+          if (data.data_output[0]['data_type'] == MimeTypes.IQ_SIGMF_32) {
```

Comment:
you should still be able to do:

        if (data.data_output && data.data_output.length > 0 && data.data_output[0]['data_type'] == MimeTypes.IQ_SIGMF_32) 
        
It's just the ordering of it matters as expressions are executed from left to right

Reply:
I did it this way because of the else. That code also needs to be protected by the null & array element test.